src/lib/search/trading-entities.ts

```diff
@@ -0,0 +1,33 @@
+import { writable } from "svelte/store";
+import searchClient from "./client";
+
+const searchOptions = {
+  query_by: "name,token_tickers,token_names,smart_contract_addresses",
+  sort_by: "type_rank:asc,_text_match:desc,volume_24h:desc",
+  group_by: "type"
+};
+
+const { subscribe, set } = writable([]);
+let collection;
+
+if (searchClient) {
+  collection = searchClient.collections("trading-entities").documents();
+}
+
+async function search(query) {
```

Comment:
Can you add a comment here about this function / module what does it do? Maybe have `query` as typed.